optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/SubSingleBenchmarkResult.java

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.benchmark.impl.result;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamOmitField;
+import org.optaplanner.benchmark.impl.measurement.ScoreDifferencePercentage;
+import org.optaplanner.benchmark.impl.report.BenchmarkReport;
+import org.optaplanner.benchmark.impl.statistic.ProblemStatistic;
+import org.optaplanner.benchmark.impl.statistic.PureSingleStatistic;
+import org.optaplanner.benchmark.impl.statistic.SingleStatistic;
+import org.optaplanner.benchmark.impl.statistic.StatisticType;
+import org.optaplanner.core.api.score.FeasibilityScore;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.impl.score.ScoreUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents 1 benchmark run for 1 Single Benchmark configuration for 1 {@link Solver} configuration for 1 problem
+ * instance (data set).
+ */
+@XStreamAlias("subSingleBenchmarkResult")
+public class SubSingleBenchmarkResult implements SolverProblemBenchmarkResult {
+
+    protected static final transient Logger logger = LoggerFactory.getLogger(SubSingleBenchmarkResult.class);
+
+    @XStreamOmitField
+    private SingleBenchmarkResult singleBenchmarkResult;
+
+    private int subSingleBenchmarkIndex = -1;
+
+    // @XStreamImplicit() // TODO FIXME
+    private List<PureSingleStatistic> subPureSingleStatisticList = null;
```

Comment:
Doesn't really make sense to create additional Statistics, as they are the same as before. If you meant only renaming them, well, that doesn't sound like an improvement either, as that would make a mess in SingleBenchmarkResult.

+1 that we could somehow decide to store the statistics differently: right now, they're copied from the parent Single when the SubSingles are created, and upon accumulating, the median's statistics are copied back to the parent. See [this JIRA comment](https://issues.jboss.org/browse/PLANNER-401?focusedCommentId=13105187&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13105187) proposal D2c. (Proposal: move D4 into a new JIRA - we could add those charts Jirka is talking about, but separately - maybe)